californium-core/src/test/java/org/eclipse/californium/core/test/ObserveTest.java

```diff
@@ -205,6 +205,52 @@ public void onError() { }
 		
 	}
 	
+	/**
+	 * Test case for CoapClient.observeAndWait(Request request, CoapHandler handler).
+	 * @author Praful Bhatnagar (prafulbhatnagar@gmail.com)
+	 * @throws Exception
+	 */
+	@Test
+	public void testObserveAndWait() throws Exception {
```

Comment:
I think we are already (implicitly) testing the existing behavior of `CoapClient.observeAndWait` by means of the `testObserve` test case, aren't we? In this test case we should probably only make sure that we detect an illegal argument, i.e. a `Request` without the *observe* option set.